test/Mono.Linker.Tests.Cases/DataFlow/ApplyTypeAnnotations.cs

```diff
@@ -123,5 +125,45 @@ class FromStringConstantWithAnnotationTestType
 			string typeName)
 		{
 		}
+
+		// Issue: https://github.com/mono/linker/issues/1537
+		//[Kept]
+		//[KeptMember (".ctor()")]
+		class FromStringConstantWithGenericInner
+		{
+		}
+
+		[Kept]
+		[KeptMember (".ctor()")]
+		class FromStringConstantWithGeneric<T>
+		{
+			[Kept]
+			public T GetValue () { return default (T); }
+		}
+
+		[Kept]
+		static void TestFromStringConstantWithGeneric ()
+		{
+			RequireCombinationOnString ("Mono.Linker.Tests.Cases.DataFlow.ApplyTypeAnnotations+FromStringConstantWithGeneric`1[[Mono.Linker.Tests.Cases.DataFlow.ApplyTypeAnnotations+FromStringConstantWithGenericInner]]");
```

Comment:
Do we have a test with assembly name which does not exist or is invalid?

Reply:
I added them - but they're not really testing much - we intentionally ignore unresolved type names, so it's really just testing that we don't crash on it.